### PR TITLE
Add chunking to CERTIFICATE response

### LINKED ISSUE #1018: support CHUNK_GET for CERTIFICATE
According to 10.8 GET_CERTIFICATE, the Responder shall fragment and serve CERTIFICATE message in chunks.

https://github.com/DMTF/libspdm/blob/e8c3632e1edcc9588714e84cb154705e8c908e97/library/spdm_responder_lib/libspdm_rsp_certificate.c#L114-L123

Current implementation does not support chunks.

### PR DESCRIPTION
**Pull request by steven-bellock**
Fixes #1018.

Still needs unit test coverage.

Signed-off-by: Steven Bellock <sbellock@nvidia.com>

### CODE REVIEW

File: library/spdm_responder_lib/libspdm_rsp_certificate.c [hunk 1]
```diff
@@ -104,46 +103,49 @@ libspdm_return_t libspdm_get_response_certificate(void *context,
...
 
-    libspdm_reset_message_buffer_via_request_code(spdm_context, session_info,
-                                                  spdm_request->header.request_response_code);
+    if (!libspdm_is_capabilities_flag_supported(spdm_context, false,
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_CHUNK_CAP,
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_CHUNK_CAP)) {
+        if (length > LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN) {
+            length = LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN;
+        }
+    }
```
Comment (jyao1) [inline library/spdm_responder_lib/libspdm_rsp_certificate.c:132]: Why you add check cap check for LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN ?

Comment (jyao1) [reply library/spdm_responder_lib/libspdm_rsp_certificate.c:132]: Is there any test to cover the new case: (CHECK_CAP == 1) && (length > LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN) ?

Comment (steven-bellock) [reply library/spdm_responder_lib/libspdm_rsp_certificate.c:132]: > Why you add check cap check for LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN ?

If chunking is not supported then it reverts to the old behavior.

Comment (jyao1) [reply library/spdm_responder_lib/libspdm_rsp_certificate.c:132]: I am thinking below:
1) If LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN is ignored in CHUNK, then we should describe in spdm_config.h.
2) If we want to change, we should also change in encap_certificate.
3) We need have new test to cover this new corner case, to ensure it still works in all scenario.

Comment (steven-bellock) [reply library/spdm_responder_lib/libspdm_rsp_certificate.c:132]: Looking through the code I don't think `LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN` is being used correctly / consistently. For example `libspdm_get_certificate` uses `LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN` as the maximum size of the entire certificate, rather than just a portion. It's possible it's not needed anymore now that we have the send and receive buffers. I'll file a separate issue.

Comment (jyao1) [reply library/spdm_responder_lib/libspdm_rsp_certificate.c:132]: > For example libspdm_get_certificate uses LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN as the maximum size of the entire certificate, rather than just a portion.

Would you please point out where the code is? I think length means portion_length
```
    length = LIBSPDM_MIN(length, LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN);
```
Why it is entire certificate?

**Comment by steven-bellock** [comment]
The https://github.com/DMTF/libspdm/blob/main/unit_test/test_spdm_requester/chunk_get.c#L552 test needs to be overhauled. We can discuss that in the meeting tomorrow.

**Comment by jyao1** [comment]
question: Have you tested spdm-emu, with chunk enabled?

**Comment by steven-bellock** [comment]
> question: Have you tested spdm-emu, with chunk enabled?

I synced to head of `spdm-emu` and ran `spdm_requester_emu` and `spdm_responder_emu` and they worked.
